src/ir_Coolix.cpp

```diff
@@ -215,125 +275,144 @@ void IRCoolixAC::off(void) {
   this->setPower(false);
 }
 
-bool IRCoolixAC::getSwing() { return remote_state == kCoolixSwing; }
+bool IRCoolixAC::getSwing() { return swingCounter>0; }
 
-void IRCoolixAC::setSwing() {
-  // Assumes that repeated sending "swing" toggles the action on the device.
-  updateSavedState();
-  remote_state = kCoolixSwing;
+void IRCoolixAC::setSwing(const bool sendflag) {
+  if(++swingCounter==0){
+    swingCounter=1;
+  }
+  if(sendflag)
+    send(kCoolixSwing,kCoolixBits);
 }
 
-bool IRCoolixAC::getSleep() { return remote_state == kCoolixSleep; }
+bool IRCoolixAC::getSwingH() { return swingHCounter>0; }
 
-void IRCoolixAC::setSleep() {
-  updateSavedState();
-  remote_state = kCoolixSleep;
+void IRCoolixAC::setSwingH(const bool sendflag) {
+  if(++swingHCounter==0){
+    swingHCounter=1;
+  }
+  if(sendflag)
+    send(kCoolixSwingH,kCoolixBits);
 }
 
-bool IRCoolixAC::getTurbo() { return remote_state == kCoolixTurbo; }
+bool IRCoolixAC::getSwingV() { return swingVCounter>0; }
 
-void IRCoolixAC::setTurbo() {
-  // Assumes that repeated sending "turbo" toggles the action on the device.
-  updateSavedState();
-  remote_state = kCoolixTurbo;
+void IRCoolixAC::setSwingV(const bool sendflag) {
+  if(++swingVCounter==0){
+    swingVCounter=1;
```

Comment:
Curious: Why are you using a counter to track this rather than a `bool`? Help me understand.